lib/versions.js

```diff
@@ -1,22 +1,22 @@
+import fs from "fs";
 import * as core from "@actions/core";
 import { HttpClient } from "@actions/http-client";
 
 export async function determineVersion(repo, versionInput) {
-  versionInput = versionInput.trim();
+  versionInput = versionInput?.trim();
 
   if (versionInput === "latest") {
     versionInput = await fetchLatestTag(repo);
   }
 
-  if (versionInput.startsWith("v")) {
-    versionInput = versionInput.substring(1);
-  }
-
   if (!versionInput) {
-    throw new Error(`'scarb-input' value must not be empty`);
+    let toolVersion = getVersionFromToolVersionsFile();
+    versionInput = toolVersion ?? (await fetchLatestTag(repo));
   }
 
-  return versionInput;
+  return versionInput.startsWith("v")
+    ? versionInput.substring(1)
+    : versionInput;
 }
```

Comment:
I believe this doesn't work if user does not specify `scarb-version` and has no `.tool-versions` file. We still should print a meaningful error in this case, telling the user what is wrong and how to fix it (I now see that it was missing, but please add it)

Reply:
Ok, @maciektr  pointed out to me that this works as expected. Ignore this comment.

Reply:
We can just have `versionInput = toolVersion ?? "latest"` and move the if after this one for readability, but this is a super-nitpick